src/snowflake/snowpark/modin/pandas/series.py

```diff
@@ -2752,8 +2752,6 @@ def _update_inplace(self, new_query_compiler):
         # Propagate changes back to parent so that column in dataframe had the same contents
         if self._parent is not None:
             if self._parent_axis == 0:
-                self._parent.loc[self.name] = self
```

Comment:
why this failed. Is this a bug in loc?

Reply:
Yup - seems like a loc bug:
```python
In [1]: import pandas as pd

In [2]: df = pd.DataFrame([[1, 2, 3], [4, None, 6]], columns=list("ABC"))

In [3]: df.loc[1]
Out[3]:
A    4.0
B    NaN
C    6.0
Name: 1, dtype: float64

In [4]: df.loc[1] = df.loc[1].fillna(method="ffill")
<ipython-input-4-e4cfb7fcf01d>:1: FutureWarning: Series.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
  df.loc[1] = df.loc[1].fillna(method="ffill")

In [5]: df
Out[5]:
   A    B  C
0  1  2.0  3
1  4  4.0  6

In [6]: im();
snow
In [7]: snow_df = pd.DataFrame([[1, 2, 3], [4, None, 6]], columns=list("ABC"))

In [8]: snow_df.loc[1] = snow_df.loc[1].fillna(method="ffill")
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[8], line 1
----> 1 snow_df.loc[1] = snow_df.loc[1].fillna(method="ffill")

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/pandas/indexing.py:1000, in _LocIndexer.__setitem__(self, key, item)
    998     item = pd.DataFrame(item)
    999 item = item._query_compiler if isinstance(item, BasePandasDataset) else item
-> 1000 new_qc = self.qc.set_2d_labels(
   1001     index,
   1002     columns,
   1003     item,
   1004     matching_item_columns_by_label=matching_item_columns_by_label,
   1005     matching_item_rows_by_label=matching_item_rows_by_label,
   1006     index_is_bool_indexer=index_is_bool_indexer,
   1007 )
   1009 self.df._update_inplace(new_query_compiler=new_qc)

File ~/.miniconda3/envs/snowpark/lib/python3.10/site-packages/modin/logging/logger_decorator.py:125, in enable_logging.<locals>.decorator.<locals>.run_and_log(*args, **kwargs)
    110 """
    111 Compute function with logging if Modin logging is enabled.
    112
   (...)
    122 Any
    123 """
    124 if LogMode.get() == "disable":
--> 125     return obj(*args, **kwargs)
    127 logger = get_logger()
    128 logger.log(log_level, start_line)

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:9267, in SnowflakeQueryCompiler.set_2d_labels(self, index, columns, item, matching_item_columns_by_label, matching_item_rows_by_label, index_is_bool_indexer, deduplicate_columns)
   9264         row_frame = get_index_frame_by_row_label_slice(self._modin_frame, index)
   9265         index = SnowflakeQueryCompiler(row_frame)
-> 9267 result_frame = set_frame_2d_labels(
   9268     internal_frame=self._modin_frame,
   9269     index=index._modin_frame
   9270     if isinstance(index, SnowflakeQueryCompiler)
   9271     else index,
   9272     columns=columns,
   9273     item=item._modin_frame
   9274     if isinstance(item, SnowflakeQueryCompiler)
   9275     else item,
   9276     matching_item_columns_by_label=matching_item_columns_by_label,
   9277     matching_item_rows_by_label=matching_item_rows_by_label,
   9278     index_is_bool_indexer=index_is_bool_indexer,
   9279     deduplicate_columns=deduplicate_columns,
   9280 )
   9282 return SnowflakeQueryCompiler(result_frame)

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py:2238, in set_frame_2d_labels(internal_frame, index, columns, item, matching_item_columns_by_label, matching_item_rows_by_label, index_is_bool_indexer, deduplicate_columns)
   2227 if item_is_frame:
   2228     # when item is not frame, this map will be initialized later
   2229     item_data_col_label_to_pos_map = {
   2230         label: pos
   2231         for pos, label in enumerate(
   (...)
   2235         )
   2236     }
-> 2238     result_frame = _set_2d_labels_helper_for_frame_item(
   2239         internal_frame,
   2240         index,
   2241         item,
   2242         matching_item_columns_by_label,
   2243         matching_item_rows_by_label,
   2244         col_info,
   2245         index_is_bool_indexer,
   2246     )
   2248 if item_is_scalar:
   2249     result_frame = _set_2d_labels_helper_for_non_frame_item(
   2250         internal_frame, index, index_is_bool_indexer
   2251     )

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py:1775, in _set_2d_labels_helper_for_frame_item(internal_frame, index, item, matching_item_columns_by_label, matching_item_rows_by_label, col_info, index_is_bool_indexer)
   1772     actual_num_cols_item = len(item.data_column_pandas_labels)
   1773     if expected_num_cols_item != actual_num_cols_item:
   1774         # when matching item by position, pandas requires the number of columns match between df and item
-> 1775         raise ValueError(
   1776             LOC_SET_ITEM_SHAPE_MISMATCH_ERROR_MESSAGE.format(
   1777                 actual_num_cols_item, expected_num_cols_item
   1778             )
   1779         )
   1781 if index_is_bool_indexer:
   1782     result_frame = align_on_index(
   1783         internal_frame,
   1784         index,
   1785         "coalesce",
   1786     ).result_frame

ValueError: shape mismatch: the number of columns 1 from the item does not match with the number of columns 3 to set
```

Reply:
But there's also a bigger bug, where we don't seem to realize that _parent_axis is 1 in the above case:
```python
In [9]: snow_df.loc[1]._parent_axis
Out[9]: 0

In [10]: snow_df.loc[:, 'B']._parent_axis
Out[10]: 0
```
so really I think this is a broader indexing error.

Reply:
```python
In [12]: snow_df.loc[1] = pd.Series([4, 4, 5], index=list("ABC"))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[12], line 1
----> 1 snow_df.loc[1] = pd.Series([4, 4, 5], index=list("ABC"))

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/pandas/indexing.py:1000, in _LocIndexer.__setitem__(self, key, item)
    998     item = pd.DataFrame(item)
    999 item = item._query_compiler if isinstance(item, BasePandasDataset) else item
-> 1000 new_qc = self.qc.set_2d_labels(
   1001     index,
   1002     columns,
   1003     item,
   1004     matching_item_columns_by_label=matching_item_columns_by_label,
   1005     matching_item_rows_by_label=matching_item_rows_by_label,
   1006     index_is_bool_indexer=index_is_bool_indexer,
   1007 )
   1009 self.df._update_inplace(new_query_compiler=new_qc)

File ~/.miniconda3/envs/snowpark/lib/python3.10/site-packages/modin/logging/logger_decorator.py:125, in enable_logging.<locals>.decorator.<locals>.run_and_log(*args, **kwargs)
    110 """
    111 Compute function with logging if Modin logging is enabled.
    112
   (...)
    122 Any
    123 """
    124 if LogMode.get() == "disable":
--> 125     return obj(*args, **kwargs)
    127 logger = get_logger()
    128 logger.log(log_level, start_line)

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py:9267, in SnowflakeQueryCompiler.set_2d_labels(self, index, columns, item, matching_item_columns_by_label, matching_item_rows_by_label, index_is_bool_indexer, deduplicate_columns)
   9264         row_frame = get_index_frame_by_row_label_slice(self._modin_frame, index)
   9265         index = SnowflakeQueryCompiler(row_frame)
-> 9267 result_frame = set_frame_2d_labels(
   9268     internal_frame=self._modin_frame,
   9269     index=index._modin_frame
   9270     if isinstance(index, SnowflakeQueryCompiler)
   9271     else index,
   9272     columns=columns,
   9273     item=item._modin_frame
   9274     if isinstance(item, SnowflakeQueryCompiler)
   9275     else item,
   9276     matching_item_columns_by_label=matching_item_columns_by_label,
   9277     matching_item_rows_by_label=matching_item_rows_by_label,
   9278     index_is_bool_indexer=index_is_bool_indexer,
   9279     deduplicate_columns=deduplicate_columns,
   9280 )
   9282 return SnowflakeQueryCompiler(result_frame)

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py:2238, in set_frame_2d_labels(internal_frame, index, columns, item, matching_item_columns_by_label, matching_item_rows_by_label, index_is_bool_indexer, deduplicate_columns)
   2227 if item_is_frame:
   2228     # when item is not frame, this map will be initialized later
   2229     item_data_col_label_to_pos_map = {
   2230         label: pos
   2231         for pos, label in enumerate(
   (...)
   2235         )
   2236     }
-> 2238     result_frame = _set_2d_labels_helper_for_frame_item(
   2239         internal_frame,
   2240         index,
   2241         item,
   2242         matching_item_columns_by_label,
   2243         matching_item_rows_by_label,
   2244         col_info,
   2245         index_is_bool_indexer,
   2246     )
   2248 if item_is_scalar:
   2249     result_frame = _set_2d_labels_helper_for_non_frame_item(
   2250         internal_frame, index, index_is_bool_indexer
   2251     )

File ~/Documents/snowpark-python/src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py:1775, in _set_2d_labels_helper_for_frame_item(internal_frame, index, item, matching_item_columns_by_label, matching_item_rows_by_label, col_info, index_is_bool_indexer)
   1772     actual_num_cols_item = len(item.data_column_pandas_labels)
   1773     if expected_num_cols_item != actual_num_cols_item:
   1774         # when matching item by position, pandas requires the number of columns match between df and item
-> 1775         raise ValueError(
   1776             LOC_SET_ITEM_SHAPE_MISMATCH_ERROR_MESSAGE.format(
   1777                 actual_num_cols_item, expected_num_cols_item
   1778             )
   1779         )
   1781 if index_is_bool_indexer:
   1782     result_frame = align_on_index(
   1783         internal_frame,
   1784         index,
   1785         "coalesce",
   1786     ).result_frame

ValueError: shape mismatch: the number of columns 1 from the item does not match with the number of columns 3 to set

In [13]: native_df = native_pd.DataFrame([[1, 2, 3], [4, None, 6]], columns=list("ABC"))

In [14]: native_df.loc[1] = native_pd.Series([4, 4, 5], index=list("ABC"))

In [15]: native_df
Out[15]:
   A    B  C
0  1  2.0  3
1  4  4.0  5
```

Reply:
Should I open issues for these two bugs? cc: @sfc-gh-vbudati

Reply:
Good catch since `.loc[self.name]` only looks at row labels but `[self.name]` checks the column labels

Reply:
locset issue: https://snowflakecomputing.atlassian.net/browse/SNOW-1649172
Series parent axis issue: https://snowflakecomputing.atlassian.net/browse/SNOW-1649176

Reply:
@sfc-gh-rdurrani I think the right fix should set `_parent_axis` correctly but not remove the code here.

Reply:
Makes sense to me! I'm putting up a PR for this as we speak - will draft and then remove this PR!